### PR TITLE
fix(diregapic): pipe transport flag through Bazel, add Bazel REGAPIC integration test [ggp]

### PR DESCRIPTION
**Pull request by miraleung**

### CODE REVIEW

File: rules_php_gapic/php_gapic.bzl [hunk 1]
```diff
@@ -61,15 +61,17 @@ def php_gapic_srcjar(
...
     if transport == None:
         transport = "grpc+rest"
     if transport == "grpc":
         fail("Error: gRPC-only PHP GAPIC libraries are not yet supported")
     if transport != "grpc+rest" and transport != "rest":
         fail("Error: Only 'grpc+rest' or 'rest' transports are supported")
-
-    # Generate the gapic_metadata.json file.
-    plugin_args = ["metadata"]
+    plugin_args.append("transport=%s" % transport)
```
Comment (vam-google) [inline rules_php_gapic/php_gapic.bzl:74]: Why was this moved? It is not not a big deal, but it seems like the `transport` ifs block does not have impact on this value, so it is usually better to keep modifications to the same variable (`plugin_args` in this case) together (line 65 and line 74 in this case).

File: tests/Integration/BUILD.bazel [hunk 2]
```diff
@@ -77,6 +78,27 @@ php_gapic_assembly_pkg(
...
+    deps = [
+        ":asset_rest_php_grpc",
+        ":asset_rest_php_proto",
+    ],
+)
+
+php_gapic_assembly_pkg(
+    name = "google-cloud-asset-rest-v1-php",
+    deps = [
+        ":asset_php_grpc",
```
Comment (vam-google) [inline tests/Integration/BUILD.bazel:96]: Should the rest client still depend on grpc stubs?

**Review by vam-google — APPROVED** [review]
LGTM (just a few minor comments).

File: rules_php_gapic/php_gapic.bzl (hunk 1 above)
Comment (miraleung) [reply rules_php_gapic/php_gapic.bzl:74]: The move allows us to group the `transport` setting logic together more tightly. Either has its merits, so I've moved it back.
